pynamodb/attributes.py

```diff
@@ -861,7 +933,14 @@ def deserialize(self, values):
         """
         if not self.is_raw():
             # If this is a subclass of a MapAttribute (i.e typed), instantiate an instance
-            instance = type(self)()
+            cls = type(self)
+            discriminator_attr = cls._get_discriminator_attribute()
+            if discriminator_attr:
+                discriminator_attribute_value = values.pop(discriminator_attr.attr_name, None)
+                if discriminator_attribute_value:
```

Comment:
I guess an empty string isn't valid either

Reply:
hmm, actually empty strings are valid in dynamo now for non-key attributes -- will change to a `is not None` check